File: ezbud-core/src/main/java/dk/michaelbui/ezbud/account/model/Transaction.java
package dk.michaelbui.ezbud.account.model;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

public class Transaction {
    private OffsetDateTime date;
    private Payee payee;
    private String notes;
    private String category;
    private BigDecimal payment;
    private BigDecimal deposit;
    private boolean cleared;

    public OffsetDateTime getDate() {
        return date;
    }

    public void setDate(OffsetDateTime date) {
        this.date = date;
    }

    public Payee getPayee() {
        return payee;
    }

    public void setPayee(Payee payee) {
        this.payee = payee;
    }

    public String getNotes() {
        return notes;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public BigDecimal getPayment() {
        return payment;
    }

    public void setPayment(BigDecimal payment) {
        this.payment = payment;
    }

    public BigDecimal getDeposit() {
        return deposit;
    }

    public void setDeposit(BigDecimal deposit) {
        this.deposit = deposit;
    }

    public boolean isCleared() {
        return cleared;
    }

    public void setCleared(boolean cleared) {
        this.cleared = cleared;
    }
}

File: src/EzBud.Domain/Transaction.cs
namespace EzBud.Domain;

public class Transaction(
    DateTimeOffset date,
    string payee,
    string category,
    decimal payment,
    decimal deposit,
    bool cleared = false)
{
    public DateTimeOffset Date { get; set; } = date;
    public string Payee { get; set; } = payee;
    public string? Notes { get; set; }
    public string Category { get; set; } = category;
    public decimal Payment { get; set; } = payment;
    public decimal Deposit { get; set; } = deposit;
    public bool Cleared { get; set; } = cleared;
}